homeassistant/components/decora_wifi/config_flow.py

```diff
@@ -0,0 +1,121 @@
+        if errors:
+            # Re-show the form
+            return self.async_show_form(
+                step_id="user_validate",
+                data_schema=vol.Schema(
+                    {
+                        vol.Required(
+                            CONF_USERNAME, default=self.data[CONF_USERNAME]
+                        ): str,
+                        vol.Required(CONF_PASSWORD): str,
+                    }
+                ),
+                errors=errors,
+            )
+        return await self.async_step_user_finish()
+
+    async def async_step_user_finish(self) -> FlowResult:
+        """Finish the user config flow."""
+        return self.async_create_entry(
+            title=f"{CONF_TITLE} - {self.data[CONF_USERNAME]}", data=self.data
+        )
+
+    async def async_step_reauth(self, data) -> FlowResult:
+        """Begin flow to re-authenticate an existing decora_wifi config."""
+        self.data = dict(data)
+        # Proceed to validation
+        return self.async_show_form(
+            step_id="reauth_validate",
+            data_schema=vol.Schema({vol.Required(CONF_PASSWORD): str}),
+        )
+
+    async def async_step_reauth_validate(self, user_input) -> FlowResult:
+        """Validate reauth with user-provided password and request re-entry if validation fails."""
+        # Update password record stored in data
+        self.data[CONF_PASSWORD] = user_input[CONF_PASSWORD]
+        errors = await self._async_validate_credentials()
+        if errors:
+            # Re-show the form
+            return self.async_show_form(
+                step_id="reauth_validate",
+                data_schema=vol.Schema({vol.Required(CONF_PASSWORD): str}),
+                errors=errors,
+            )
+        return await self.async_step_reauth_finish()
+
+    async def async_step_reauth_finish(self) -> FlowResult:
+        """Finish the reauth config flow."""
+        entry = await self.async_set_unique_id(self.data[CONF_USERNAME].lower())
+        if entry:
+            self.hass.config_entries.async_update_entry(
+                entry, title=f"{CONF_TITLE} - {CONF_USERNAME}", data=self.data
+            )
+            await self.hass.config_entries.async_reload(entry.entry_id)
+        return self.async_abort(reason="reauth_successful")
+
+    async def _async_validate_credentials(self) -> dict[str, str]:
+        """Call the myLeviton API to validate user-provided credentials."""
+        session = None
+        errors: dict[str, str] = {}
+        try:
+            session = await DecoraWifiPlatform.async_setup_decora_wifi(
+                self.hass,
+                email=self.data[CONF_USERNAME],
+                password=self.data[CONF_PASSWORD],
+            )
+        except LoginFailed:
+            errors["base"] = "invalid_auth"
+        except CommFailed:
+            errors["base"] = "cannot_connect"
+        if session:
+            session.teardown()
```

Comment:
Based on `async_unload_entry` it looks like this needs to be run in the executor